pkg/sql/session.go

```diff
@@ -368,12 +370,12 @@ func (ts *txnState) resetStateAndTxn(state TxnStateEnum) {
 // The session context is just used for logging the SQL trace.
 func (ts *txnState) finishSQLTxn(sessionCtx context.Context) {
 	ts.mon.Stop(ts.Ctx)
-	if ts.sp == nil {
-		panic("No span in context? Was resetForNewSQLTxn() called previously?")
```

Comment:
@andreimatei This panic implies this shouldn't happen, yet looking at `resetForNewSQLTxn` it seems that it always occurs if `traceSQL` is true. Am I missing something?